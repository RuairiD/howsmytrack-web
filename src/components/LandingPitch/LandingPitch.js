import React from 'react';

import { Icon, List, Typography } from 'antd';

type Props = {
    isMobile: boolean,
};

class LandingPitch extends React.Component<Props> {
    /*
     * Component for displaying FAQ page explaining what How's My Track? is etc.
     */

    steps = [
        {
            icon: "plus",
            text: "Submit your track.",
        },
        {
            icon: "usergroup-add",
            text: "Join a group.",
        },
        {
            icon: "edit",
            text: "Write feedback for your peers.",
        },
        {
            icon: "check",
            text: "Read your own feedback.",
        },
        {
            icon: "rollback",
            text: "Repeat.",
        },
    ]

    render() {
        return (
            <div style={{ textAlign: 'center' }}>
                {this.props.isMobile && <div>
                    <img alt="how's my track" src="/logo512.png" width="200px" style={{ marginBottom: '2em'}} />
                </div>}
                {!this.props.isMobile && <Typography.Title level={1} style={{ marginBottom: '1em' }}>
                    What is <em>how's my track?</em>
                </Typography.Title>}
                <List
                    style={{ display: "inline-block", textAlign: 'left' }}
                    itemLayout="vertical"
                    dataSource={this.steps}
                    renderItem={step => (
                        <Typography.Title level={3} style={{ display: 'flex' }}>
                            <Icon type={step.icon} style={{ marginRight: '2em' }} />
                            <div>{step.text}</div>
                        </Typography.Title>
                    )}
                />
            </div>
        );
    }
}

export default LandingPitch;