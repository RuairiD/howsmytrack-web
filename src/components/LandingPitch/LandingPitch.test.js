import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { act } from "react-dom/test-utils";
import LandingPitch from "./LandingPitch";

jest.mock("react-ga");

describe("LandingPitch", () => {
    it("renders login and register buttons, along with an explanation of how the site works", () => {
        const wrapper = shallow(
            <LandingPitch />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("shows a LoginModal when Sign In button is clicked, and hides it when dismissed", () => {
        const wrapper = mount(
            <LandingPitch />,
        );

        act(() => wrapper.find("Button").get(1).props.onClick());

        wrapper.update();
        expect(wrapper.find("LoginModal").get(0).props.isVisible).toBeTruthy();

        act(() => wrapper.find("LoginModal").get(0).props.onCancel());
        wrapper.update();

        expect(wrapper.find("LoginModal").get(0).props.isVisible).not.toBeTruthy();
    });

    it("shows a RegisterModal when Register button is clicked, and hides it when dismissed", () => {
        const wrapper = mount(
            <LandingPitch />,
        );

        act(() => wrapper.find("Button").get(0).props.onClick());

        wrapper.update();
        expect(wrapper.find("RegisterModal").get(0).props.isVisible).toBeTruthy();

        act(() => wrapper.find("RegisterModal").get(0).props.onCancel());
        wrapper.update();

        expect(wrapper.find("RegisterModal").get(0).props.isVisible).not.toBeTruthy();
    });
});
