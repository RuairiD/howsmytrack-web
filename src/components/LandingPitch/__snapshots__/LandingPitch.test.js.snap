// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPitch renders login and register buttons, along with an explanation of how the site works 1`] = `
<Div
  className=""
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <Div
    className=""
  >
    <Img
      alt="how's my track"
      className=""
      src="/logo512.png"
      style={
        Object {
          "marginBottom": "2em",
        }
      }
      width="256px"
    />
  </Div>
  <Div
    className=""
    style={
      Object {
        "display": "flex",
        "margin": "1em",
        "textAlign": "center",
      }
    }
  >
    <Button
      block={false}
      className="home-action-button"
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      size="large"
      style={
        Object {
          "marginLeft": "auto",
        }
      }
    >
      Register
    </Button>
    <Button
      block={false}
      className="home-action-button"
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      size="large"
      style={
        Object {
          "marginRight": "auto",
        }
      }
    >
      Sign In
    </Button>
  </Div>
  <List
    bordered={false}
    dataSource={
      Array [
        Object {
          "icon": "plus",
          "text": "Submit your track.",
        },
        Object {
          "icon": "usergroup-add",
          "text": "Join a group.",
        },
        Object {
          "icon": "edit",
          "text": "Write feedback for your peers.",
        },
        Object {
          "icon": "check",
          "text": "Read your own feedback.",
        },
        Object {
          "icon": "rollback",
          "text": "Repeat.",
        },
      ]
    }
    itemLayout="vertical"
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
    style={
      Object {
        "display": "inline-block",
        "textAlign": "left",
      }
    }
  />
  <LoginModal
    isVisible={false}
    onCancel={[Function]}
  />
  <RegisterModal
    isVisible={false}
    onCancel={[Function]}
  />
</Div>
`;
